test/integration/src/test/java/io/pravega/test/integration/WatermarkingTest.java

```diff
@@ -0,0 +1,245 @@
+        @Cleanup
+        ReaderGroupManager watermarkingRG = new ReaderGroupManagerImpl(scope, controller, syncClientFactory, connectionFactory);
+        String markRG = "watermarkingRG";
+        String markStream = NameUtils.getMarkStreamForStream(stream);
+        watermarkingRG.createReaderGroup(markRG, ReaderGroupConfig.builder().stream(Stream.of(scope, markStream)).build());
+
+        // create reader
+        @Cleanup
+        final EventStreamReader<Watermark> markReader = clientFactory.createReader("markreader",
+                markRG,
+                new WatermarkSerializer(),
+                ReaderConfig.builder().build());
+        EventRead<Watermark> eventRead = markReader.readNextEvent(30000L);
+        while (eventRead.getEvent() == null) {
+            eventRead = markReader.readNextEvent(30000L);
+        }
+        Watermark watermark = eventRead.getEvent();
+        stopFlag.set(true);
+
+        // read events from the stream
+        @Cleanup
+        ReaderGroupManager readerGroupManager = new ReaderGroupManagerImpl(scope, controller, syncClientFactory, connectionFactory);
+        String readerGroup = "rg";
+
+        long timeLow = watermark.getLowerTimeBound();
+        Map<Segment, Long> positionMap0 = watermark.getStreamCut()
+                                                   .entrySet().stream().collect(Collectors.toMap(x -> new Segment(scope, stream, x.getKey().getSegmentId()),
+                        Map.Entry::getValue));
+
+        StreamCut streamCutStart = new StreamCutImpl(streamObj, positionMap0);
+        Map<Stream, StreamCut> start = Collections.singletonMap(streamObj, streamCutStart);
+        
+        readerGroupManager.createReaderGroup(readerGroup, ReaderGroupConfig.builder().stream(streamObj)
+                                                                           .startingStreamCuts(start).build());
+
+        // create reader on the stream
+        @Cleanup
+        final EventStreamReader<String> reader = clientFactory.createReader("myreader",
+                readerGroup,
+                javaSerializer,
+                ReaderConfig.builder().build());
+        
+        // read events from the reader. 
+        // verify that events read belong to the bound
+        EventRead<String> event = reader.readNextEvent(10000L);
+        while (event.getEvent() != null) {
+            Long time = Long.parseLong(event.getEvent());
+            assertTrue(time >= timeLow);
+            event = reader.readNextEvent(10000L);
+        }
+    }
+
+    private void scale(Controller controller, Stream streamObj, StreamConfiguration configuration) throws InterruptedException {
+        // perform several scales
+        int numOfSegments = configuration.getScalingPolicy().getMinNumSegments();
+        double delta = 1.0 / numOfSegments;
+        for (long segmentNumber = 0; segmentNumber < numOfSegments - 1; segmentNumber++) {
+            Thread.sleep(5000L);
```

Comment:
Theses sorts of long delays are appropriate for a System test but not an integration test. This is going to run on every commit and the whole of the `gradle check` must run in under 1 hour on travis. So we want to keep them relatively quick.